### PR TITLE
Present predictions as prometheus metrics

### PR DESCRIPTION
**Pull request by dbason**
This PR exposes the Arima predictions as prometheus metrics.  Currently we use the data stored in elasticsearch as history for the model, so I have left this in until we move to using Thanos for historical data.

### CODE REVIEW

File: service-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  name: opni-metrics-service
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: opni-svc-metrics
+      app.kubernetes.io/part-of: opni
+      opni.io/service: metrics
+  endpoints:
+  - port: metrics
+---
```
Comment (tybalex) [inline service-monitor.yaml:34]: just wondering , what does this yaml file do?

File: metric-forecasting/metric_streaming.py [hunk 2]
```diff
@@ -61,6 +66,62 @@
...
+async def get_metrics():
+    metrics_payloads = []
+    for metric_name, query in PROMETHEUS_CUSTOM_QUERIES.items():
+        current_data = prom.custom_query(query=query)
+        prediction = MAD_DICT[metric_name].run(current_data)
+        if prediction:
+            metrics_payloads.append(prediction)
+            for column in COLUMNS_LIST:
+                if prediction[column] is not None:
+                    GAUGE_DICT[metric_name].labels(value_type=column).set(prediction[column])
```
Comment (tybalex) [inline metric-forecasting/metric_streaming.py:107]: do we need the GAUGE_DICT ? I don't find it being  used anywhere else. is it reserved for Thanos ?

File: service-monitor.yaml (hunk 1 above)
Comment (dbason) [reply service-monitor.yaml:34]: This file sets up the prometheus resources to make sure the metrics service predictions get scraped.

File: metric-forecasting/metric_streaming.py (hunk 2 above)
Comment (dbason) [reply metric-forecasting/metric_streaming.py:107]: Yeah, the Gauge objects in the dictionary are instantiated in the startup
```
async def startup_event():
    logger.info("loading history on webserver startup")
    for metric_name in PROMETHEUS_CUSTOM_QUERIES.keys():
        MAD_DICT[metric_name] = MetricAnomalyDetector(metric_name)
        GAUGE_DICT[metric_name] = Gauge(metric_name, metric_name, ['value_type'])
        await load_history_data(metric_name, MAD_DICT[metric_name])
```
Then the method that actually writes out the prometheus metrics makes use of them
```
generate_latest(REGISTRY).decode('utf-8')
```

It's all part of the prometheus-client library

**Review by tybalex — APPROVED** [review]
look good. should we merge it now or wait till the Thanos integration?

**Comment by dbason** [comment]
I think we should get this in first.  It's still using Elasticsearch for the historical data so we can switch that over when Thanos is implemented.

**Review by AmartC — APPROVED** [review]

**Pull request merged**